packages/playout-gateway/src/tsrHandlerJobs/jobQueue.ts

```diff
@@ -0,0 +1,231 @@
+		importance?: JobImportance,
+		id?: string
+	): IChainableJobQueue<T> {
+		this.queue.push({ id, job, chained: false, timeout, importance, abortController: new AbortController() })
+		this.run()
+		return this as IChainableJobQueue<T>
+	}
+
+	chain<T, U, V>(
+		job: Job<T, U, V>,
+		timeout?: number,
+		importance?: JobImportance,
+		id?: string
+	): IChainableJobQueue<T> {
+		this.queue.push({ id, job, chained: true, timeout, importance, abortController: new AbortController() })
+		return this as IChainableJobQueue<T>
+	}
+
+	end<X>(onFulfilled?: FulfilledCallback<X>, onRejected?: RejectedCallback, onTimeout?: TimeoutCallback): void {
+		const lastJob = this.queue[this.queue.length - 1]
+		if (!lastJob) return
+		lastJob.onFulfilled = onFulfilled
+		lastJob.onRejected = onRejected
+		lastJob.onTimeout = onTimeout
+	}
+
+	getLastJobIdByImportance(importance: JobImportance): string | undefined {
+		for (let i = this.queue.length - 1; i >= 0; --i) {
+			if (this.queue[i].importance === importance) {
+				return this.queue[i].id
+			}
+		}
+		if (this.currentJob?.importance === importance) {
+			return this.currentJob.id
+		}
+		return undefined
+	}
+
+	removeJob(jobId: string): void {
+		this.queue = this.queue.filter((job) => job.id !== jobId && !job.chained)
+	}
+
+	endsWith(jobId: string): boolean {
+		const lastJob = this.queue[this.queue.length - 1] ?? this.currentJob
+		return lastJob?.id === jobId
+	}
+
+	private run(): void {
+		setImmediate(() => {
+			if (this.currentJob) {
+				return
+			}
+			if (this.queue.length === 0 && this.currentJobChain.length === 0) {
+				this.emit('done')
+			}
+			this.currentJob = this.queue.shift() ?? null
+			if (!this.currentJob) return
+			const jobChainTop = this.currentJobChain[this.currentJobChain.length - 1]
+			const prevResult = jobChainTop?.result
+			const currentJob = this.currentJob
+			if (this.currentJob.timeout) {
+				this.currentJobTimeout = setTimeout(() => {
+					this.abortJob(currentJob)
+					currentJob.didTimeOut = true
+				}, this.currentJob.timeout)
+			}
+			this.currentJob.job
+				.run(prevResult, this.currentJob.abortController.signal)
+				.then(this.handleJobFulfilled)
+				.catch(this.handleJobRejected)
+				.then(this.handleCleanupResults)
+				.finally(this.advanceJob)
+		})
+	}
+
+	private handleJobFulfilled = (result: any): void => {
+		if (!this.currentJob) return
+		if (this.currentJob.abortController.signal.aborted) {
+			throw new AbortError()
+		}
+		this.currentJob.result = result
+		const nextJob = this.queue[0]
+		if (nextJob?.chained) {
+			this.currentJobChain.push(this.currentJob)
+		}
+		if (this.currentJob.onFulfilled) {
+			this.currentJob.onFulfilled(this, result)
+		}
+	}
+
+	private handleJobRejected = async (reason: any): Promise<PromiseSettledResult<void>[] | void> => {
+		if (!this.currentJob) return
+		this.logger.error(
+			`Error in queue "${this.name}" at Job.run "${this.currentJob.job.constructor.name}": ${reason}`
+		)
+		const ps = []
```

Comment:
This might just be me, but abbreviations on arrays make it hard to follow the general idea of having a descriptive name in plural. I know Rasmus' made comments about the abbreviation for promises as well on the related PR.

Reply:
Not just you, i think it was Anders who commented on a similar abbreviations, somewhere else.
Some abbreviations can be ok, in my mind, if it is otherwise a very long name, as long as understanding of what it is, is not lost. - this is not such a case.